Modules/arraymodule.c

```diff
@@ -197,20 +197,22 @@ b_setitem(arrayobject *ap, Py_ssize_t i, PyObject *v)
     /* PyArg_Parse's 'b' formatter is for an unsigned char, therefore
        must use the next size up that is signed ('h') and manually do
        the overflow checking */
-    if (!PyArg_Parse(v, "h;array item must be integer", &x))
+    if (!PyArg_Parse(v, "h:array('b').__setitem__", &x)) {
```

Comment:
actually, after the patch, it would be `OverflowError: Python int too large to convert to C short`, because the `h` specifier is used in the call to PyArg_Parse()...